src/charm.py

```diff
@@ -0,0 +1,256 @@
+#!/usr/bin/env python3
+
+import os
+import subprocess
+import socket
+import re
+import pwd
+import sys
+sys.path.append('lib')  # noqa
+
+from ops.charm import CharmBase, CharmEvents
+from ops.framework import EventBase, EventSource, StoredState
+from ops.main import main
+from ops.model import (
+    ActiveStatus,
+    BlockedStatus,
+    MaintenanceStatus,
+    WaitingStatus,
+    ModelError,
+)
+from cluster import CockroachDBCluster
+from interface_proxy_listen_tcp import ProxyListenTcpInterfaceProvides
+
+from jinja2 import Environment, FileSystemLoader
+from datetime import timedelta
+from time import sleep
+
+
+class CockroachStartedEvent(EventBase):
+    pass
+
+
+class ClusterInitializedEvent(EventBase):
+    def __init__(self, handle, cluster_id):
+        super().__init__(handle)
+        self.cluster_id = cluster_id
+
+    def snapshot(self):
+        return self.cluster_id
+
+    def restore(self, cluster_id):
+        self.cluster_id = cluster_id
+
+
+class CockroachDBCharmEvents(CharmEvents):
+    cockroachdb_started = EventSource(CockroachStartedEvent)
+    cluster_initialized = EventSource(ClusterInitializedEvent)
```

Comment:
Everything here is from CochroachDB, so either we consistently prefix them after the database, which doesn't seem so nice since the class itself already gives these names context, or we avoid the term in all of them. My preference is for the latter.

The first one might be _database_started_, or _daemon_started_, etc, depending on what it means.

Reply:
Agreed, `daemon_started` or even `daemonized` would be better.